execution_engine/src/core/engine_state/mod.rs

```diff
@@ -442,9 +445,45 @@ where
             tracking_copy.borrow_mut().force_write(*key, value.clone());
         }
 
+        // Perform global state migrations that require state.
+
+        if let Some(activation_point) = upgrade_config.activation_point() {
+            // The highest stored era is the immediate predecessor of the activation point.
+            let highest_era_info_id = activation_point.saturating_sub(1);
+            let highest_era_info_key = Key::EraInfo(highest_era_info_id);
+
+            let get_result = tracking_copy
+                .borrow_mut()
+                .get(correlation_id, &highest_era_info_key)
+                .map_err(|error| Error::Exec(error.into()))?;
+
+            match get_result {
+                Some(stored_value @ StoredValue::EraInfo(_)) => {
+                    tracking_copy
+                        .borrow_mut()
+                        .force_write(Key::EraSummary, stored_value);
+                }
+                Some(other_stored_value) => {
+                    // This should not happen as we only write EraInfo variants.
+                    error!(stored_value_type_name=%other_stored_value.type_name(),
+                        "EraInfo key contains unexpected StoredValue variant");
+                    return Err(Error::ProtocolUpgrade(
+                        ProtocolUpgradeError::UnexpectedStoredValueVariant,
+                    ));
+                }
+                None => {
+                    // Can't find key
+                    // Most likely this chain did not yet ran an auction, or recently completed a
+                    // prune
```

Comment:
I don't think this comment is accurate, and while it is basically harmless to do so I don't think it is necessary to log this circumstance as a `warn`. 

As far as I recall, a new network w/ genesis at this binary or higher would never have such an entry (correct behavior). An existing network upgraded into this binary _must_ have such an entry. In a future binary we will either remove this handling altogether, or if it is left in place it would essentially translate to noop on networks that had already previously gone thru a migration. 

Am i forgetting something?

Reply:
Your comment is correct, the `warn!` here was useful for testing early in the development to get the math right. I removed the warning.